src/clr-angular/data/tree-view/models/tree-node.model.ts

```diff
@@ -55,6 +83,18 @@ export abstract class TreeNodeModel<T> {
     this.setSelected(newState, true, propagate);
   }
 
+  // To bypass the default event handlers and programmatically set the checked' & 'indeterminate' DOM properties of input checkbox
+  checkboxToggleHandler(propagate: boolean, event: Event) {
```

Comment:
I don't see a need for this new method instead of just adding this logic into the existing `toggleSelection` method, can you merge them so we don't have two public methods here?